src/obu.rs

```diff
@@ -549,6 +541,47 @@ fn parse_seq_hdr(c: &mut Rav1dContext, gb: &mut GetBits) -> Rav1dResult<Rav1dSeq
     })
 }
 
+pub(crate) unsafe fn rav1d_parse_sequence_header(
+    mut data: &[u8],
+) -> Rav1dResult<DRav1d<Rav1dSequenceHeader, Dav1dSequenceHeader>> {
+    while !data.is_empty() {
+        let mut gb = GetBits::new(data);
+
+        gb.get_bit(); // obu_forbidden_bit
+        let raw_type = gb.get_bits(4);
+        let r#type = Rav1dObuType::from_repr(raw_type as usize);
+        let has_extension = gb.get_bit();
+        let has_length_field = gb.get_bit();
+        gb.get_bits(1 + has_extension as i32 * 8); // reserved
+
+        // obu length field
+        let obu_end = if has_length_field {
+            let len = gb.get_uleb128() as usize;
+            if len > data.len() {
+                return Err(EINVAL);
+            }
+            len
+        } else {
+            data.len()
+        };
+
+        if r#type == Some(Rav1dObuType::SeqHdr) {
+            if let Ok(seq_hdr) = parse_seq_hdr(&mut gb, false) {
+                return Ok(DRav1d::from_rav1d(seq_hdr));
+            }
```

Comment:
```suggestion
            let res = parse_seq_hdr(&mut gb, false);
            res?;
```

Reply:
I removed the `let` to avoid creating a shadow variable